src/Workspaces/Core/Portable/Workspace/Solution/ProjectDependencyGraph_RemoveProject.cs

```diff
@@ -73,12 +77,15 @@ private static ImmutableDictionary<ProjectId, ImmutableHashSet<ProjectId>> Compu
             // 'projectId' from the set of references if it exists.
             foreach (var (id, _) in existingForwardReferencesMap)
             {
-                builder.MultiRemove(id, removedProjectId);
+                foreach (var removedProjectId in removedProjectIds)
+                    builder.MultiRemove(id, removedProjectId);
             }
         }
 
         // Finally, remove 'projectId' itself.
-        builder.Remove(removedProjectId);
+        foreach (var removedProjectId in removedProjectIds)
+            builder.Remove(removedProjectId);
```

Comment:
You could call RemoveRange.

Reply:
ImmutableDictionary's AddRange takes in an IEnumerable, and thus allocates the enumerator. I figured it was pretty trivial to just do the loop at this level.

Reply:
My assumption was ImmutableDictionary could optimize further and avoid extra allocations than doing it one at a time to offset the enumerator allocations, but I see if this is a builder and not the ImmutableDictionary itself then:

https://github.com/dotnet/runtime/blob/d2c561859bd6474b8ec38bf4ee7296e4e717ce1c/src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableDictionary_2.Builder.cs#L439-L450

Maybe then add a comment at each site that we're not using RemoveRange to avoid the enumerator allocation?